src/control/jobs/control_jobs.c

```diff
@@ -2060,7 +2060,38 @@ void dt_control_write_sidecar_files()
                                                           FALSE));
 }
 
+static gboolean _same_basefilename(const char *filename, const char *prev_filename)
+{
+  if(!filename || !prev_filename) return FALSE;
+  const char *dot = strrchr(filename, '.');
+  if(!dot) return FALSE;
+  const int length = dot - filename;
+  const char *prev_dot = strrchr(prev_filename, '.');
+  if(!prev_dot) return FALSE;
+  const int prev_length = prev_dot - prev_filename;
+  if(length != prev_length)
+    return FALSE;
+  for(int i = length - 1; i > 0; i--)
+    if(filename[i] != prev_filename[i])
+      return FALSE;
+  return TRUE;
+}
+
+static char *_set_extension(const char *prev_output, const char *filename)
```

Comment:
Same here, I'll move that into utility.h.

Also I would use `g_malloc()`.

Reply:
done